### DISCUSSION TITLE
Show the CreateAction in ListRecords only if $table is empty?

### DISCUSSION THREAD

**Discussion by devhoussam1998** [filamentphp/filament#9817] [Help] [answered]
I want to show the **getHeaderActions** **CreateAction** in my **ListCustomers** only if there is no records in $table?

**ListCustomers.php**

```
<?php

...

class ListCustomers extends ListRecords
{
    protected static string $resource = CustomerResource::class;

    protected function getHeaderActions(): array
    {
        return [
            Actions/CreateAction::make()
        ];
    }
}
```

**CustomerResource.php**

```
<?php

...

class CustomerResource extends Resource
{
   ...
   
   public static function table(Table $table): Table
   {
        return $table
        ->columns([
            
        ])
        ->filters([
            
        ])
        ->actions([
            
        ])
        ->bulkActions([
            
        ]);
    }
}
```

**Comment by coolsam726** [comment]
```php
Actions\CreateAction::make()->visible(fn() => !($this->table?->getQuery()?->count() > 0)),
```

**Reply by devhoussam1998** [reply]
<img width="1422" alt="Screenshot 2023-11-24 at 12 57 03" src="https://github.com/filamentphp/filament/assets/121255432/c3bf09ac-0566-420e-ac03-45a9ac4d7eeb">

as you can see in picture above I have two `CreateAction` the first one should be displayed only when table has records and the second one inside the table it will displayed when there is no records on table.

**Reply by coolsam726** [reply]
Oh in that case just negate the above condition for the first action and you should achieve your goal. Let me know if I am missing something.

**Reply by devhoussam1998** [reply]
How?

**Reply by coolsam726** [reply]
Actions\CreateAction::make()->visible(fn() => $this->table?->getQuery()?->count() > 0),

**Reply by coolsam726** [reply, marked as answer]
```php
Actions\CreateAction::make()->visible(fn() => $this->table?->getQuery()?->count() > 0),
```
Sorry, I'm posting from my phone.

**Reply by devhoussam1998** [reply]
thank you @coolsam726
